### PR TITLE
Add Azure PR/branch jobs in prow

### PR DESCRIPTION
**Pull request by 0xmichalis**
Depends on https://github.com/openshift/aos-cd-jobs/pull/1301 + https://github.com/openshift/aos-cd-jobs/pull/1294

@pschiffe @jim-minter @kwoodson

### CODE REVIEW

File: cluster/ci/config/prow/config.yaml [hunk 1]
```diff
@@ -379,6 +379,32 @@ presubmits:
...
+    agent: jenkins
+    labels:
+      master: ci.openshift.redhat.com
+    always_run: true
+    # TODO: Remove once the job is stable
+    skip_report: true
+    branches:
+    - release-3.9
+    context: ci/openshift-jenkins/azure
+    rerun_command: "/test azure"
```
Comment (kargakis) [inline cluster/ci/config/prow/config.yaml]: @pweil- you had raised discussion about the experience around launching these jobs, does this lgty?

Comment (pweil-) [reply cluster/ci/config/prow/config.yaml]: :+1:

**Comment by kargakis** [comment]
I hope this won't saturate our quota

/hold cancel

**Pull request merged**